### PR TITLE
feature/injection

### PR DESCRIPTION
**Pull request by MoatazBadawy**

### CODE REVIEW

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,39 +1,20 @@
...
-        <intent>
-            <action android:name="com.google.android.youtube.api.service.START" />
-        </intent>
-    </queries>
-    <queries>
-        <package android:name="com.android.chrome" />
-    </queries>
 
     <application
+        android:name=".injection.app.AfternoonHadithApp"
```
Comment (MahmoudMabrok) [inline app/src/main/AndroidManifest.xml:10]: i think `app` class should not be at injection package

File: app/src/main/java/com/moataz/afternoonhadeeth/data/remote/model/videos/top/TopVideosBlocks.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.moataz.afternoonhadeeth.data.remote.model.videos.top
+
+import androidx.annotation.Keep
+
+@Keep
+data class TopVideosBlocks(
+    val itemFour: com.moataz.afternoonhadeeth.data.remote.model.videos.top.ItemFour?,
+    val itemOne: com.moataz.afternoonhadeeth.data.remote.model.videos.top.ItemOne?,
+    val itemThree: com.moataz.afternoonhadeeth.data.remote.model.videos.top.ItemThree?,
+    val itemTwo: com.moataz.afternoonhadeeth.data.remote.model.videos.top.ItemTwo?
```
Comment (MahmoudMabrok) [inline app/src/main/java/com/moataz/afternoonhadeeth/data/remote/model/videos/top/TopVideosBlocks.kt:10]: i think we need to revisit naming here

File: app/src/main/java/com/moataz/afternoonhadeeth/data/repository/BooksRepository.kt [hunk 3]
```diff
@@ -1,20 +1,20 @@
...
+import com.moataz.afternoonhadeeth.data.remote.request.ApiClint
 import com.moataz.afternoonhadeeth.utils.helper.HttpRoutes
 import com.moataz.afternoonhadeeth.utils.status.Resource
 import io.ktor.client.request.*
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.flow
 import kotlinx.coroutines.flow.flowOn
 
-class BooksRepository : BooksAPIService {
+class BooksRepository : com.moataz.afternoonhadeeth.data.remote.api.BooksAPIService {
```
Comment (MahmoudMabrok) [inline app/src/main/java/com/moataz/afternoonhadeeth/data/repository/BooksRepository.kt:13]: Repo should extend repo interface & accept sources.

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (MoatazBadawy) [reply app/src/main/AndroidManifest.xml:10]: But the Application class we shouldn't add to any specific package name? Instead, it is alone outsides other packages?

File: app/src/main/java/com/moataz/afternoonhadeeth/data/remote/model/videos/top/TopVideosBlocks.kt (hunk 2 above)
Comment (MoatazBadawy) [reply app/src/main/java/com/moataz/afternoonhadeeth/data/remote/model/videos/top/TopVideosBlocks.kt:10]: Great, I changed it in the last update, this was an old version from API.

File: app/src/main/java/com/moataz/afternoonhadeeth/data/repository/BooksRepository.kt (hunk 3 above)
Comment (MoatazBadawy) [reply app/src/main/java/com/moataz/afternoonhadeeth/data/repository/BooksRepository.kt:13]: Could you check out the app repo after the last commit, and review whether it meets your comment tip or not now?

**Pull request merged**